pkg/guacrest/openapi.yaml

```diff
@@ -28,39 +28,108 @@ paths:
+  "/analysis/dependencies":
     get:
-      summary: Retrieve transitive dependencies 
-      description: > 
-        Find the transitive dependencies of the input. The HasSBOM and HasSLSA
-        predicates are used as the dependency relationship and the IsOccurrence and 
-        PkgEqual predicates are used to find consider equivalent packages.
-      operationId: retrieveDependencies
+      summary: Identify the most important dependencies
+      operationId: analyzeDependencies
       parameters:
         - $ref: "#/components/parameters/PaginationSpec"
-        - name: linkCondition
-          description: > 
-            Whether links between nouns must be made by digest or if they 
-            can be made just by name (i.e. purl). Specify 'name' to allow using
-            SBOMs that don't provide the digest of the subject. The default is 
-            'digest'. To search by purl, 'name' must be specified.
+        - name: sort
+          description: >
+            The sort order of the packages
+              * 'frequency' - The packages with the highest number of dependents
+              * 'scorecard' - The packages with the lowest OpenSSF scorecard score
           in: query
-          required: false
+          required: true
           schema:
-            type: string 
-            enum: 
-              - digest
-              - name
+            type: string
+            enum:
+              - frequency
+              - scorecard
+      responses:
+        "200":
+          $ref: "#/components/responses/PackageNameList"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+        "502":
+          $ref: "#/components/responses/BadGateway"
+  "/v0/package/{purl}":
+    get:
+      summary: Get all purls related to the given purl
+      description: >
+        If a partial purl, pkg:foo/bar is passed in, then it would find all purls 
+        associated with the given purl, such as pkg:foo/bar@1.0, and pkg:foo/bar@2.0.
+      operationId: getPackagePurls
+      parameters:
         - name: purl
-          description: The purl of the dependent package.
-          in: query
-          required: false
+          in: path
+          required: true
+          description: URL-encoded Package URL (purl)
           schema:
             type: string
-        - name: digest
-          description: The digest of the dependent package.
+      responses:
+        "200":
+          $ref: "#/components/responses/PurlList"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+        "502":
+          $ref: "#/components/responses/BadGateway"
+  "/v0/package/{purl}/vulns":
+    get:
+      summary: Get vulnerabilities for a Package URL (purl)
+      description: >
+        If a partial purl is passed in, then it will find all associated purls 
```

Comment:
See the other comment on using partial vs precise purls in these endpoints.

Reply:
Done